pkgs/top-level/all-packages.nix

```diff
@@ -5717,6 +5717,8 @@ in
 
   knockknock = callPackage ../tools/security/knockknock { };
 
+  knockpy = python3Packages.callPackage ../tools/security/knockpy { };
```

Comment:
```suggestion
  knockpy = callPackage ../tools/security/knockpy { };
```

Reply:
What's wrong with using `python3Packages.callPackage` here?

`python3Packages.callPackage` allows to access Python parts and packages directly. Some people are requesting to use it during reviews for a while now.

It's a CLI tool and there is no module, thus using `toPythonApplication` wouldn't not help. Using `python3Packages.callPackage` allows to quickly change it to a Python package, if that it added/needed in the future, as the differences between `buildPythonApplication` and `buildPythonPackage` would be minimal.

Reply:
It doesn't allow overriding Python packages the right way.

Reply:
Ah, you were talking about the dependencies and not `knockpy`. 

It looks like that a little over 100 packages are using `python3Packages.callPackage`. It's interesting that this is not causing issues in a larger scale. Could be that all those packages are well-maintained.